language-support/hs/bindings/src/DA/Ledger/Services/MeteringReportService.hs

```diff
@@ -128,41 +70,52 @@ timestampToIso8601 ts = ISO8601.iso8601Show ut
     st = timestampToSystemTime ts
     ut = System.systemToUTCTime st
 
-isoTimeToTimestamp :: IsoTime -> Timestamp
-isoTimeToTimestamp iso = systemTimeToTimestamp $ System.utcToSystemTime $ unIsoTime iso
-
 utcDayToTimestamp :: Day -> Timestamp
 utcDayToTimestamp day = systemTimeToTimestamp $ System.utcToSystemTime $ UTCTime day (secondsToDiffTime 0)
 
-raiseApplicationMeteringReport :: LL.ApplicationMeteringReport -> Perhaps MeteredApplication
-raiseApplicationMeteringReport (LL.ApplicationMeteringReport llApp events) = do
-  application <- raiseApplicationId llApp
-  return MeteredApplication {..}
+s2a :: Maybe S.Value -> A.Value
+s2a (Just v) = toRawAesonValue v
+s2a Nothing = A.Null
+
+a2s :: A.Value -> Maybe S.Value
```

Comment:
Here the [fromJSON](https://hackage.haskell.org/package/aeson-2.1.0.0/docs/Data-Aeson.html#v:toJSON) could be used since the instance `FromJSON` is defined and autogenerated in `Google.Protobuf.Struct.hs`.

Reply:
From discussion w/ @simonmaxen-da, it was stated that these auto generated instances add noisy metadata. So this way of providing `Aeson.Value`s was not followed.